frontend/src/helpers/index.ts

```diff
@@ -391,6 +416,82 @@ export function getHeadlampAPIHeaders(): { [key: string]: string } {
   };
 }
 
+// Export function to store cluster kubeconfig
+export function storeStatelessClusterKubeconfig(kubeconfig: string) {
+  // Get existing stored cluster kubeconfigs
+  const storedClusterKubeconfigsJSON = sessionStorage.getItem('clusterKubeconfigs');
+
+  let storedClusterKubeconfigs: string[] = [];
+  if (storedClusterKubeconfigsJSON) {
+    storedClusterKubeconfigs = JSON.parse(storedClusterKubeconfigsJSON);
+  }
+
+  // Check if the kubeconfig is not already in the array, then add it
+  if (!storedClusterKubeconfigs.includes(kubeconfig)) {
+    storedClusterKubeconfigs.push(kubeconfig);
+  }
+
+  // Store the updated cluster kubeconfigs back in sessionStorage
+  sessionStorage.setItem('clusterKubeconfigs', JSON.stringify(storedClusterKubeconfigs));
+}
+
+// Export function to retrieve cluster kubeconfig
+export function getStatelessClusterKubeConfigs(): string[] {
+  // Get stored cluster kubeconfigs
+  const storedClusterKubeconfigsJSON = sessionStorage.getItem('clusterKubeconfigs');
+
+  if (storedClusterKubeconfigsJSON) {
+    const storedClusterKubeconfigs: string[] = JSON.parse(storedClusterKubeconfigsJSON);
+
+    return storedClusterKubeconfigs;
+  }
+
+  return [];
+}
+
+export function findKubeconfigByClusterName(clusterName: string) {
+  // Get stored cluster kubeconfigs
+  const storedClusterKubeconfigsJSON = sessionStorage.getItem('clusterKubeconfigs');
+
+  if (storedClusterKubeconfigsJSON) {
+    const storedClusterKubeconfigs: string[] = JSON.parse(storedClusterKubeconfigsJSON);
+
+    // Check each kubeconfig for the target clusterName
+    for (const kubeconfig of storedClusterKubeconfigs) {
+      const kubeconfigObject = jsyaml.load(atob(kubeconfig)) as KubeconfigObject;
+
+      // Check if the kubeconfig contains a cluster with the target clusterName
+      const matchingKubeconfig = kubeconfigObject.clusters.find(
+        cluster => cluster.name === clusterName
+      );
+      if (matchingKubeconfig) {
+        // Encode the kubeconfig back to base64 if needed
+        const encodedKubeconfig = btoa(JSON.stringify(kubeconfigObject));
+        return encodedKubeconfig;
+      }
+    }
+  }
+
+  return null; // If not found
+}
+
+export function getUserId(): string {
+  // Retrieve userId ID from sessionStorage
+  let userId = sessionStorage.getItem('userId');
```

Comment:
It's not clear to me why sessionStorage is being used here. Maybe the documentation would explain this.

Consider adding a "headlamp-" prefix to prevent possible clashes.

Reply:
Added the comment why this function is needed.